dnssec.go

```diff
@@ -221,6 +249,10 @@ func (rr *RRSIG) Sign(k PrivateKey, rrset []RR) error {
 		return ErrKey
 	}
 
+	if !isValidRRSet(rrset) {
```

Comment:
If you know for sure when you call Sign that you have a proper RRset this test does not add anything. I rather still leave this out of Sign itself.